.github/workflows/ci.yml

```
@@ -2,7 +2,16 @@ name: CI
 
 on:
   push:
+    branches:
+      - main
+      - ci-proposal
   pull_request:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'The branch, tag or SHA to checkout'
+        required: false
```

Nit: [think require = false is the default](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputsinput_idrequired) anyway so could omit. 

might be worth be explicit about what the default is:
```suggestion
        default: main
```